src/services/db_project_handler.py

```diff
@@ -61,11 +61,10 @@ def get(self, project_id: int, db: Session):
             return project_repr
 
     
-    def get_all(self, db: Session):
-        all_project_ids = db.execute(select(Projects.id))
+    def get_all(self, db: Session, accessible_projects: list[int]):
         all_projects = list()
-        for row in all_project_ids:
-            curr_project = self.get(row[0], db)
+        for project_id in accessible_projects:
+            curr_project = self.get(int(project_id), db)
             all_projects.append(curr_project)

```

Comment:
Well, this approach is workable... And I see why you do it, cuz you have logic of combining the info about the documents as well in `get` method.
Buut there is a nuance. Let's assume your specific user has access to 100 projects. Then you need to do 100 queries to get the info about each project. That's insane amount of requests to DB. If we are talking about raw SQL, tt seems that such manipulations can be done with JOINs of documents, access, user tables. But yes, of course, if we  use ORM, then somehow this joins should be done on ORM level

When you have free time (someday...), you can also read about N+1 problem. https://docs.sentry.io/product/issues/issue-details/performance-issues/n-one-queries/